### ISSUE TITLE
Preset trading tags

### ISSUE THREAD

**Issue by Tecfan** [bartervg/barter.vg#150]
# What problem does this feature address?
For tedious tasks, like adding the following trading tags to my wishlist items, a "preset trading tag" button/filler would be useful. Example:

I got the latest Monthly games, but I want extra copies to trade. I have to do the following:
Add +extra
Add Humble gift link trading tag
Add Indiegala gift link trading tag
Add Groupees gift link trading tag
Add △ trading tier

This is 5 steps which takes a long time (other people might have other "default" trading tags). I do this from item pages as my wishlist won't load. This wouldn't be too much work if I did it in my wishlist page as I could do it with 10's or 100's of games at the same time, but it won't load anyway.

## Describe a solution
A "preset" offer button that you can click on "/item/" pages that adds all 5 (x) trading tags with 1 click. I don't know how to configure this button, maybe we can have a special item that we can copy trading tags from.

Barter test item 3

Copy trading tags from -> Barter test item 3

or:

Add the same trading tags as the last edit you did.

At the very least, can "gift link" trading tags be added to the "store" trading tag select list like every other trading tag? This would double my tagging speed for extra wishlist items.

**Comment by Tecfan** [comment]
Has this been given any thought? I want to start refining my wishlist, but the current inefficiency in editing tags is stopping me. 

I might just write a userscript that adds a couple of "preset buttons", but they would require 4 or 5 reloads of the item page before completing, so it's not optimal. My eyes would hurt from all the refreshing as well x(.

Again, the goal is to do mass-addition of tags with 1 button. 

Here are some mockups... (these options/buttons has to be configured from somewhere):

Form dropdowns
![image](https://user-images.githubusercontent.com/956878/87095470-12112e00-c241-11ea-9b64-bbfdb29e5017.png)

Big buttons (3 possible placements)
![image](https://user-images.githubusercontent.com/956878/87095760-9c599200-c241-11ea-8f06-87b715685c9b.png)

Click one of these buttons, and your selected wishlist entry gets a preset set of tags, saving power users a lot of time.

**Comment by bartervg** [comment]
I fear what it would look like with UI for preset tags. 

What is an example preset tag that you want? I can skip the creating UI and add the preset tag into the usual tag menu and then concentrate on how to implement the rest.

**Comment by Tecfan** [comment]
TLDR at the bottom.

So my first todo is to add "+extra"'s to most non-tier 1 Humble games. Not taking other factors into account, my wishlist tag for those games would look like this:

Store: Humble gift link, #10 (my default sorting for wishlist, gives me some room to go up and down when refining) and "Humble gift link" trading tag (I only want gift links for extras). This is something I could do with one operation.

Then after this, I would add the "+ extra" tag. If I add "+ extra" and THEN add a Store, it looks like the "+extra" disappears, but I'm not completely sure. Anyway, "+ extra", and submit.

Then I would default by adding a "mid tier" tier tag on it, this is important. This is right now a third form submit.

And lastly, I would add, if it has been bundled or sold at IG/Groupees, "Indiegala gift link" and "Groupees gift link" tags. This is another 2 operations. Now that it has been confirmed that even more sites support gift linking, maybe it should be combated with a more general "+ extra gift" trading tag, as proposed [here](https://discordapp.com/channels/329829406153375755/438406560222281738/725782895452618773):   

> also, I just noticed that the "gift link" trading tags are not present in the offer window, which makes them kind of useless
I just got an offer for like 7 extras, but they are potentially HB keys instead of HB links, despite me having tagged them all with "humble gift link"
maybe a solution is to have two different "+extra" tags? "+extra" and "+extra gift" (does not exclude inventory gift nor gift link). i don't know. can be very cluttered if we have "ig gift link", "hb gift link", tier tag, "groupees gift link" and steam inventory gift icon on several items in an offer screen

### TLDR: 
With the current system, an example trading tag I want is (low or mid tier):
![image](https://user-images.githubusercontent.com/956878/87129367-82e03680-c291-11ea-898d-8ff6498ea61f.png)
Also proposed: a more general "gift link" trading tag to avoid clutter – let Store icon dictate what you prefer.

**Comment by bartervg** [comment]
![image](https://user-images.githubusercontent.com/28576132/87373434-b2c06000-c557-11ea-991c-3d9f11908bca.png)

I began planning a new system with pseudo tags that would refer to a different database table that would map the pseudo tag to multiple existing tags and attributes, but I think everything you described, and the addition of `gift link`, will work in the existing system. 

One problem was that when `extra` was set, it wouldn't set the store id. I think this was a fix to a previous issue. I modified this so that it will set extra and store if the store isn't 0.

**Comment by Tecfan** [comment]
Thank you, I have been using it a little. Sadly, after the great gift link controversy I don't put "gift link" in too many wishlist items anymore.

Anyway, I also made this very simple userscript that I use a lot these days:

![image](https://i.imgur.com/LV2R2TV.gif)


It just adds mid tier wishlist and mid tier notification, and it speeds up my work a lot.
